### PR TITLE
feat: provision bot channel

### PR DESCRIPTION
**Pull request by natalgar**
## Description

bot channel provisioning

## Task Item

refs #7629

### CODE REVIEW

File: extensions/azurePublishNew/src/node/azureResources/botChannel.ts [hunk 1]
```diff
@@ -15,29 +36,49 @@ export const botRegistrationDefinition: ResourceDefinition = {
...
+    const endpoint = `https://${hostname ?? config.hostname + '.azurewebsites.net'}/api/messages`;
 
-  return {
-    ...workingSet,
-    botChannelResult: provisionResult,
+    try {
+      await azureBotService.bots.create(config.resourceGroupName, config.displayName, {
+        properties: {
+          displayName: config.displayName,
+          endpoint: endpoint ?? '',
```
Comment (GeoffCoxMSFT) [inline extensions/azurePublishNew/src/node/azureResources/botChannel.ts:52]: There isn't any way for the endpoint to be undefined.  Consider checking above when constructing it.

File: extensions/azurePublishNew/src/node/azureResources/botChannel.ts [hunk 2]
```diff
@@ -15,29 +36,49 @@ export const botRegistrationDefinition: ResourceDefinition = {
...
-  const { appId } = appRegistrationResult.appId;
-  const { endpoint } = webAppResult.endpoint;
-  const { hostname } = webAppResult.hostname;
+const botChannelProvisionMethod = (provisionConfig: ProvisionConfig): ProvisionMethod => {
+  const tokenCredentials = new TokenCredentials(provisionConfig.accessToken);
+  const azureBotService = new AzureBotService(tokenCredentials, provisionConfig.subscriptionId);
 
-  const provisionResult: BotChannelResult = { botName: '' };
+  return async (config: BotServiceConfig, workingSet: ProvisionWorkingSet): Promise<ProvisionWorkingSet> => {
+    const appId = workingSet.appRegistration?.appId;
```
Comment (GeoffCoxMSFT) [inline extensions/azurePublishNew/src/node/azureResources/botChannel.ts:44]: Do you want to throw if the working set doesn't contain what this service needs?

**Comment by GeoffCoxMSFT** [comment]
Looks like you have an empty provisionService.ts file that shouldn't be added.

File: extensions/azurePublishNew/src/node/azureResources/botChannel.ts (hunk 2 above)
Comment (natalgar) [reply extensions/azurePublishNew/src/node/azureResources/botChannel.ts]: set up a new function for checking required fields

**Comment by hatpick** [comment]
I think validating the args for service calls is the job of the callers (handlers) not the service layer. Thoughts? @GeoffCoxMSFT @natalgar

**Comment by natalgar** [comment]
> I think validating the args for service calls is the responsibility of the callers (handlers) not the service layer. Thoughts? @GeoffCoxMSFT @natalgar

From OOP I tend to think of the "tell, don't ask" principle. So here I did see it as a responsibility of the service layer to validate its input.

**Comment by hatpick** [comment]
> > I think validating the args for service calls is the responsibility of the callers (handlers) not the service layer. Thoughts? @GeoffCoxMSFT @natalgar
> 
> From OOP I tend to think of the "tell, don't ask" principle. So here I did see it as a responsibility of the service layer to validate its input.

Look at it from SRP and open/closed point of view. In order to extend the validation you have to modify the service layer code. Also it's doing both service calls and validations.

**Comment by GeoffCoxMSFT** [comment]
I think if the Azure SDK is going to check the input and throw then we can rely on that and don't have to check the field and throw our own custom error.  If it is a field that we use or where we make a call and it will succeed but not do what we want we should check it.

**Review by GeoffCoxMSFT — DISMISSED** [review]

**Review by GeoffCoxMSFT — APPROVED** [review]

**Comment by natalgar** [comment]
replacing this PR with #8321

**Pull request closed without merging**